server/aesdsocket.c

```diff
@@ -257,48 +460,134 @@ int main(int argc, char *argv[])
+            }
 
-        //read full packet
-        ret = read_packet(new_socket_fd, fd);
-        if(ret == -1) { 
-            syslog(LOG_ERR, "Error reading via socket.\n");
-            socket_stat = 0;
-            fclose(fd);
-            continue;
+            ret = pthread_create(&thread, NULL, &socket_thread, t_data);
+            if(ret != 0) {
+                syslog(LOG_ERR, "Failed to create thread.\n");
+                free(t_data);
+                free(t_ptr);
+                socket_stat = 0;
+                continue;
+            }
+                
+            // addend to linked list node
+            t_ptr->thread = thread;
+            t_ptr->td = t_data;
+            SLIST_INSERT_HEAD(&head, t_ptr, entries);
         }
-        // // connection stopped
-        // if(ret == 0) { 
-        //     fclose(fd);
-        //     continue;
-        // }
-        // close file to reopen
-        fclose(fd);
 
-        // open file in read mode
-        fd = fopen(OUTPUT_FILE_PATH, "r+");
+        // timer service routine
+        if(signal_timer) {
+            signal_timer = 0; 
+            
+            //get now
+            time(&rawNow);
+            now = localtime_r(&rawNow, now);
+            
+            //format timestamp
+            memset(&data, 0, sizeof(data));
+            strftime(data, sizeof(data), "timestamp: %Y, %m, %d, %H, %M, %S\n", now);
+
+            ret = pthread_mutex_lock(&mutex);
+            if(ret != 0) {
+                syslog(LOG_ERR, "Failed to lock timestamp");
+                socket_stat = 0;
+                ret = pthread_mutex_unlock(&mutex);
+                continue;
+            }
 
-        // retransmit the file
-        ret = send_data_to_client(new_socket_fd, fd);
-        if(ret == -1) {
-            syslog(LOG_INFO, "Error sending data to client\n");
+            //write timestamp to file
+            write(fd, data, strlen(data));
+            ret = pthread_mutex_unlock(&mutex);
         }
 
-        syslog(LOG_INFO, "Closed connection from %s\n", host);
-        close(new_socket_fd); 
-        fclose(fd);
+        // check for thread status and close if complete
+        slist_thread_t* tp = NULL;
+        slist_thread_t* next = NULL;
+        SLIST_FOREACH_SAFE(tp, &head, entries, next) {
+            //check complete flag
+            if(tp->td->complete_flag == 1) {
+                //remove from linked list
+                SLIST_REMOVE(&head, tp, slist_thread_s, entries);
+                
+                //join 
+                void* t_ret = NULL;
+                ret = pthread_join(tp->thread, &t_ret);
+                if(ret != 0) {
+                    syslog(LOG_ERR, "Failed to end thread:%ld", tp->thread);
+                    socket_stat = 0;
+                }
+                
+                thread_data_t* tmp_ret = (thread_data_t *) t_ret;
+
+                //close the socket
+                syslog(LOG_DEBUG, "Closed connection from %s", host);
```

Comment:
@KapureCUB given this is a global and may be overwritten with a new connection I don't think you'll necessarily print the correct host value here.